pkg/services/grpcserver/auth_test.go

```diff
@@ -0,0 +1,88 @@
+package grpcserver
+
+import (
+	"context"
+	"testing"
+
+	apikeygenprefix "github.com/grafana/grafana/pkg/components/apikeygenprefixed"
+	"github.com/grafana/grafana/pkg/models"
+	"github.com/grafana/grafana/pkg/services/apikey"
+	"github.com/grafana/grafana/pkg/services/sqlstore"
+	"github.com/stretchr/testify/require"
+	"google.golang.org/grpc/metadata"
+)
+
+func TestAuthenticator_Authenticate(t *testing.T) {
```

Comment:
should we also add a test verifying that the new context has the user details fetched from the DB?

Reply:
added a test to confirm that the auth header was removed